pandas/core/index.py

```diff
@@ -111,9 +112,31 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None, fastpath=False,
         if fastpath:
             return cls._simple_new(data, name)
 
+        # RangeIndex pass-through
+        # Index(start, stop, ...) --> RangeIndex(start, stop, ...)
+        if isinstance(data, int):
+            if dtype is None and copy == False:
+                copy = None
+                range_constructor = True
+            elif isinstance(dtype, int):
+                range_constructor = True
+                if copy == False:
+                    copy = None
+                elif isinstance(copy, int):
+                    range_constructor = True
+                else:
```

Comment:
No real issue. It is not required for `RangeIndex`. I just thought it might be convenient for people to be able to use e.g. `Index(5)` to create a `RangeIndex(5)`.

It is kludgy since it needs to ensure, that the intent of the user really was to create a `RangeIndex` when calling `Index(...)` with a given set of parameters. I don't want to grab inputs that I shouldn't.

If the whole `Index(...)` calling options extension is an issue or just too complicated I am happy to take it out.